File: Last Stone Weight.cpp
class Solution {
public:
    int lastStoneWeight(vector<int>& stones) {
      priority_queue<int, vector<int>> maxheap;
      for(int i = 0; i<stones.size(); i++){
          maxheap.push(stones[i]);
      }
        while(maxheap.size()>=2){
            int a = maxheap.top();
            maxheap.pop();
            int b = maxheap.top();
            maxheap.pop();
            if(a>b){
                a -= b;
                maxheap.push(a);
            }
        }
        if(maxheap.size() == 0)
            return 0;
        return maxheap.top();
    }
};

File: Last Stone Weight.py
class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:
        for i in range(len(stones)):
            stones[i] *= -1
        heapify(stones)
        while len(stones) > 1:
            y = -1 * heappop(stones) 
            x = -1 * heappop(stones)
            if x != y:
                heappush(stones, -1 * abs(y-x))

        if len(stones):
            return -1 * stones[0]
        return 0
